### PR TITLE
blockdev_mirror_src_no_space: Mirror source without enough space

### PR DESCRIPTION
**Pull request by zhencliu**
Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1931333

### CODE REVIEW

**Comment by zhencliu** [comment]
virtio-scsi:
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_src_no_space.q35: PASS (210.71 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-01T02.15-ae4a34f/results.html
JOB TIME   : 213.78 s

virtio-blk:
 (1/1) Host_RHEL.m8.u3.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_mirror_src_no_space.q35: PASS (304.24 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-03-01T05.11-333a4f0/results.html
JOB TIME   : 306.89 s

**Comment by zhencliu** [comment]
hi @ngu-niny @YongxueHong , can you please help to review this patch? Thanks.

File: qemu/tests/blockdev_mirror_src_no_space.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    def check_io_error_event(self):
+        event = get_event_by_condition(
+            self.main_vm, 'BLOCK_IO_ERROR',
+            self.params.get_numeric('event_timeout', 30))
+
+        if event:
+            if event['data'].get('reason') != self.params['error_msg']:
+                self.test.fail('Unexpected error')
+        else:
```
Comment (YongxueHong) [inline qemu/tests/blockdev_mirror_src_no_space.py:24]:
```suggestion
```

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Comment by ngu-niny** [comment]
Test PASS:

 (05/60) repeat1.Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_src_no_space:  PASS (215.98 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**